docs/section10/10-4.md

```diff
@@ -0,0 +1,76 @@
+    - Get heating_from_loop (if it does not equal Null) and add to a list of heating hot water loops: `hot_water_loop_list_b = hot_water_loop_list_b.append(Terminal_b.heating_from_loop)`
+- For each boiler in B_RMR, save boiler to loop boiler dictionary: `for boiler_b in B_RMR.RulesetModelInstance.boilers: loop_boiler_dict[boiler_b.loop].append(boiler_b)`
+- For each fluid loop in B_RMR: `for fluid_loop_b in B_RMR.RulesetModelInstance.fluid_loops:`  
+    - Check if fluid loop is connected to boiler: `if fluid_loop_b in loop_boiler_dict.keys():`
+        - Check if fluid loop serves a space heating system: `if fluid_loop_b in hot_water_loop_list_b` 
+            - Add to list of applicable fluid loops: `fluid_loop_list = fluid_loop_b.append(fluid_loop_b.id)`
+- For each applicable fluid loop in fluid_loop_list: `For fluid_loop_b in fluid_loop_list:`
+    - Get boiler served by loop: `applicable_boiler = loop_boiler_dict[fluid_loop_b]`
+    - Reset energy source variable: `energy_source_type_b = ""`
+    - Get the energy_source_type: `energy_source_type_b = applicable_boiler.energy_source_type`
+    - Check if the energy source is propane, if so then set boolean variable to true: `if energy_source_type_b in ["PROPANE"]:fuels_propane_b = TRUE`
+    - Check else if the energy source is natural gas, if so then set boolean variable to true: `elif energy_source_type_b in ["NATURAL_GAS"]:fuels_natural_gas_b = TRUE`
+    - Check else if the energy source is electricity, if so then set boolean variable to true: `elif energy_source_type_b in ["ELECTRICITY"]:fuels_electricity_b = TRUE`
+    - Check else if the energy source is fuel oil, if so then set boolean variable to true: `elif energy_source_type_b in ["FUEL_OIL"]:fuels_oil_b = TRUE`
+    - Else, set other boolean variable to true: `Else:fuels_other_b = TRUE`  
+- Get whether baseline heating fuel is electricity and natural gas: `if fuels_natural_gas_b == TRUE AND fuels_electricity_b == TRUE: elec_and_natural_gas_b = TRUE`
+
+**Rule Assertion:**  
+- Case 1: if the baseline heating energy source is natural gas then pass: `if fuels_other_b == FALSE AND fuels_oil_b == FALSE AND fuels_natural_gas_b == TRUE AND fuels_electricity_b == FALSE AND fuels_propane_b == FALSE: PASS`
+- Case 2: elif the baseline heating energy source is electricity then pass: `if fuels_other_b == FALSE AND fuels_oil_b == FALSE AND fuels_natural_gas_b == FALSE AND fuels_electricity_b == TRUE AND fuels_propane_b == FALSE: PASS`
+- Case 3: elif the baseline heating energy source is anything other than electricity, propane, or natural gas: `elif fuels_other_b == TRUE OR fuels_oil_b == TRUE: FAIL`
+- Case 4: elif the baseline energy source is propane and there is natural gas in the proposed then fail: `elif fuels_natural_gas_b == FALSE AND fuels_electricity_b == FALSE AND fuels_propane_b == TRUE AND "NATURAL_GAS" is in heating_fuels_p: Fail`
```

Comment:
What if baseline is modeled with both propane AND natural gas, or propane AND electricity? I think it should also be "fail". So I think it should be 
elif fuels_electricity_b == FALSE AND fuels_propane_b == TRUE AND "NATURAL_GAS" is in heating_fuels_p: Fail`

Reply:
I updated the syntax with the logic in the first sentence. I need some help understanding how the proposed syntax accomplishes the logic in the first sentence. I did what I thought made sense, let me know if changes are needed.

Reply:
I think to accomplish the logic in the first sentense of the comment above, it should just be "if fuels_propane_b == TRUE AND "NATURAL_GAS" is in heating_fuels_p: Fail". This would address both Case 4 and Case 5.

Reply:
Shouldn't we also have a fail for when electricity and gas are modeled in the baseline?